optimum/habana/transformers/trainer.py

```diff
@@ -214,6 +214,11 @@ def __init__(
             "ignore", message="User provided device_type of 'cuda', but CUDA is not available. Disabling"
         )
 
+        # HPU graphs training optimization, where FW+BWD passes are scheduled on HPU for execution within milliseconds.
+        self._recorded_fwbwd_graph = None
+        self._recorded_fwbwd_inputs = None
+        self._recorded_fwbwd_output = None
```

Comment:
We should declare these attributes only if using HPU graphs for training, which may not be the case. I think for now we can use the argument `self.args.use_hpu_graphs` but we may need to distinguish between training and inference at some point.

Reply:
True. I will address that isssue.

Reply:
Maybe we'll want to have two separate arguments for training and inference such as `args.use_hpu_graphs_for_training` and `args.use_hpu_graphs_for_inference`. I'll discuss that with Sree and will let you know what we decide to do moving forward.